src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/UnixObjectWriter.cs

```diff
@@ -167,8 +238,12 @@ private protected override void CreateSection(ObjectNodeSection section, string
                         _dwarfEhFrame.AddFde(fde);
                     }
 
-                    EmitLsda(nodeWithCodeInfo, frameInfos, i, _lsdaSectionWriter, ref mainLsdaOffset);
+                    if (emittedLsdaSymbols == null)
+                        EmitLsda(nodeWithCodeInfo, frameInfos, i, _lsdaSectionWriter, ref mainLsdaOffset);
```

Comment:
This can be folded into the previous `if` on line 217. It was originally split to support prepending additional data in the LSDA section but that's solved in a different way for ARM32 now.